lib/factor/factorQuadratic.js

```diff
@@ -0,0 +1,127 @@
+'use strict';
+
+const checks = require('../checks');
+const ConstantFactors = require('./ConstantFactors');
+const ChangeTypes = require('../ChangeTypes');
+const flatten = require('../../lib/util/flattenOperands');
+const Negative = require('../Negative');
+const Node = require('../node');
+
+// Given a node, will check if it's in the form of a quadratic equation
+// `ax^2 + bx + c`, and
+// if it is, will factor it using on of the following rules:
+//    - Difference of squares e.g. x^2 - 4 -> (x+2)(x-2)
+//    - Perfect square e.g. x^2 + 2x + 1 -> (x+1)^2
+//    - Sum/product rule e.g. x^2 + 3x + 2 -> (x+1)(x+2)
+//    - TODO: quadratic formula
+function factorQuadratic(node) {
+  node = flatten(node);
+  if (!checks.isQuadratic(node)) {
+    return Node.Status.noChange(node);
+  }
+
+  // get a, b and c
+  let firstTermNode, secondTermNode, constantTermValue;
+  for (const term of node.args) {
+    if (Node.Type.isConstant(term)) {
+      constantTermValue = term.eval();
+    }
+    else if (Node.PolynomialTerm.isPolynomialTerm(term)) {
+      const polyTerm = new Node.PolynomialTerm(term);
+      const exponent = polyTerm.getExponentNode(true);
+      if (exponent.value === '2') {
+        firstTermNode = polyTerm;
+      }
+      else if (exponent.value === '1') {
+        secondTermNode = polyTerm;
+      }
+      else {
+        return Node.Status.noChange(node);
+      }
+    }
+    else {
+      return Node.Status.noChange(node);
+    }
+  }
+
+  if (!firstTermNode || !constantTermValue) {
+    return Node.Status.noChange(node);
+  }
+
+  const symbol = firstTermNode.getSymbolNode();
```

Comment:
ooohhh in the quadratic check can you make sure they have the same symbol? and add a test for that too